substrate/utils/frame/benchmarking-cli/src/pallet/command.rs

```diff
@@ -589,111 +582,75 @@ impl PalletCmd {
 					.build_storage()
 					.map_err(|e| format!("{ERROR_CANNOT_BUILD_GENESIS}\nError: {e}"))?;
 
-				(storage, Default::default())
+				storage
 			},
-			(Some(GenesisBuilder::Runtime), _) | (None, true) =>
-				(Default::default(), self.genesis_from_runtime::<H, F>()?),
-		})
-	}
+			(Some(GenesisBuilder::SpecRuntime), Some(_)) =>
+				return Err("Cannot use `--genesis-builder=spec` with `--runtime` since the runtime would be ignored.".into()),
+			(Some(GenesisBuilder::SpecRuntime), None) => {
+				let Some(chain_spec) = chain_spec else {
+					return Err("No chain spec specified to generate the genesis state".into());
+				};
 
-	/// Generate the genesis changeset by the runtime API.
-	fn genesis_from_runtime<H: Hash, F: HostFunctions>(&self) -> Result<OverlayedChanges<H>> {
-		let state = BenchmarkingState::<H>::new(
-			Default::default(),
-			Some(self.database_cache_size as usize),
-			false,
-			false,
-		)?;
+				self.genesis_from_spec_runtime::<F>(chain_spec.as_ref())?
+			},
+			(Some(GenesisBuilder::Runtime), None) => return Err("Cannot use `--genesis-builder=runtime` without `--runtime`".into()),
+			(Some(GenesisBuilder::Runtime), Some(runtime)) | (None, Some(runtime)) => {
+				let runtime = std::path::absolute(runtime)
+					.map_err(|e| format!("Could not get absolute path for runtime file: {e}"))?;
```

Comment:
```suggestion
```

Don't really need this. If we can make it absolute, `fs::read` should also be able to read it.